esphome/components/stepper/__init__.py

```diff
@@ -66,11 +68,22 @@ def validate_speed(value):
     return value
 
 
+Rotation = stepper_ns.enum("Rotation")
+ROTATIONS = {
+    "BOTH": Rotation.ROTATION_BOTH,
+    "CW": Rotation.ROTATION_CW,
+    "CLOCKWISE": Rotation.ROTATION_CW,
+    "CCW": Rotation.ROTATION_CCW,
+    "COUNTERCLOCKWISE": Rotation.ROTATION_CCW,
+    "COUNTER-CLOCKWISE": Rotation.ROTATION_CCW,
+}
```

Comment:
Lets not add multiple options per each type. It can add confusion and be less clear in the documentation.

```suggestion
ROTATIONS = {
    "BOTH": Rotation.ROTATION_BOTH,
    "CLOCKWISE": Rotation.ROTATION_CW,
    "COUNTERCLOCKWISE": Rotation.ROTATION_CCW,
}
```